src/urllib3/poolmanager.py

```diff
@@ -494,40 +528,29 @@ def _set_proxy_headers(self, url, headers=None):
             headers_.update(headers)
         return headers_
 
-    def _validate_proxy_scheme_url_selection(self, url_scheme):
-        if (
-            url_scheme == "https"
-            and self.proxy.scheme == "https"
-            and not self.allow_insecure_proxy
-        ):
-            warnings.warn(
-                "Your proxy configuration specified an HTTPS scheme for the proxy. "
-                "Are you sure you want to use HTTPS to contact the proxy? "
-                "This most likely indicates an error in your configuration."
-                "If you are sure you want use HTTPS to contact the proxy, enable "
-                "the _allow_https_proxy_to_see_traffic.",
-                InvalidProxyConfigurationWarning,
-            )
-
-            raise ProxySchemeUnsupported(
-                "Contacting HTTPS destinations through HTTPS proxies is not supported."
-            )
-
     def urlopen(self, method, url, redirect=True, **kw):
         "Same as HTTP(S)ConnectionPool.urlopen, ``url`` must be absolute."
         u = parse_url(url)
-        self._validate_proxy_scheme_url_selection(u.scheme)
+        self._set_destination_scheme(u.scheme)
 
-        if u.scheme == "http" or self.proxy.scheme == "https":
+        if not connection_requires_http_tunnel(self.proxy, self.proxy_config, u.scheme):
             # For connections using HTTP CONNECT, httplib sets the necessary
-            # headers on the CONNECT to the proxy. For HTTP or when talking
-            # HTTPS to the proxy, we'll definitely need to set 'Host' at the
-            # very least.
+            # headers on the CONNECT to the proxy. If we're not using CONNECT,
+            # we'll definitely need to set 'Host' at the very least.
             headers = kw.get("headers", self.headers)
             kw["headers"] = self._set_proxy_headers(url, headers)
 
         return super(ProxyManager, self).urlopen(method, url, redirect=redirect, **kw)
 
+    def _set_destination_scheme(self, destination_scheme):
```

Comment:
Can't this change between requests? I don't think `connection_pool_kw` should be modified outside the constructor. Maybe need to find another way to get the scheme down to the ConnectionPool?

Reply:
I was able to improve how the destination scheme was used within ``poolmanager.py`` and ``connectionpool.py``. Let me know what you think!